host.go

```diff
@@ -209,15 +209,43 @@ func (h *gpbftRunner) computeNextInstanceStart(cert *certs.FinalityCertificate)
 		return h.clock.Now().Add(ecDelay)
 	}
 
+	base := ts.Timestamp()
+
+	defer func() {
+		now := h.clock.Now()
+		round0Duration := 4 * h.manifest.Gpbft.Delta
+
+		// If we were supposed to start this instance more than one GPBFT round ago, assume
+		// we're behind and align our start time to the next multiple of GPBFT round. This
+		// helps keep nodes in-sync when bootstrapping and catching up.
+		if _nextStart.Before(now.Add(-round0Duration)) {
+			// Given that we're catching up, we can assume that everyone is largely in
+			// agreement and we should decide everything in round 0. Therefore, we
+			// to finish in ~4x the maximum time it takes a message to propagate across
+			// the network.
+			//
+			// Aligning on this boundary means we can complete instances as fast as
+			// possible while still staying roughly in-sync.
+			//
+			// TODO: we should consider adding a buffer (e.g., a 2x multiplier?).
```

Comment:
Seems like something to add to manifest?

Reply:
Agreed

Reply:
Added to the manifest. It shouldn't be a breaking change (json ignores unknown fields in the manifest when unmarshaling, and we default to disabled).